src/Illuminate/Queue/Queue.php

```diff
@@ -104,9 +113,15 @@ protected function createPayload($job, $data = '')
      */
     protected function createPayloadArray($job, $data = '')
     {
-        return is_object($job)
+        $payload = is_object($job)
                     ? $this->createObjectPayload($job)
                     : $this->createStringPayload($job, $data);
+
+        if (! is_null($this->metadata) && $this->metadata->isNotEmpty()) {
+            $payload['metadata'] = $this->metadata->toArray();
```

Comment:
I'm wondering if `$payload['metadata']` should *always* be set - even if it is to `null`.

Otherwise on the other end if I want to consume metadata - I'll have to check if it even exists first. Would be better to know it is always there...?

Reply:
Potentially,  I did it this way to not break backwards compatibility and only add the new key in if they wanted to use it.

Reply:
So maybe on the interface there should be a `getMetadata()`? That way you can always get the metadata via a getter, and it can check if the key even exists? That would give more reason to even have the interface in the first place - i.e. you provide a `get` and `set` - rather than just `set`?